cmd/cli/setup/update.go

```diff
@@ -0,0 +1,137 @@
+package setup
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/storacha/piri/pkg/client"
+)
+
+var SupportedLinuxArch = map[string]bool{
+	"amd64": true,
+	"arm64": true,
+}
+
+var (
+	UpdateCmd = &cobra.Command{
+		Use:   "update",
+		Args:  cobra.NoArgs,
+		Short: "Check for and apply updates to piri",
+		Long: `Check for new releases and update the piri binary to the latest version.
+
+This command downloads and installs the latest version but does not restart the service.
+To check if an update is safe, use 'piri status upgrade-check' first.`,
+		RunE: doUpdate,
+	}
+
+	checkOnly bool
+	force     bool
+	version   string
+)
+
+func init() {
+	UpdateCmd.SetOut(os.Stdout)
+	UpdateCmd.SetErr(os.Stderr)
+	UpdateCmd.Flags().BoolVar(&checkOnly, "check", false, "Check for updates without applying them")
+	UpdateCmd.Flags().BoolVar(&force, "force", false, "Skip safety checks and force update")
+	UpdateCmd.Flags().StringVar(&version, "version", "", "Update to specific version (e.g., v1.2.3)")
+}
+
+// GitHubRelease represents a GitHub release
+type GitHubRelease struct {
+	TagName    string    `json:"tag_name"`
+	Name       string    `json:"name"`
+	Draft      bool      `json:"draft"`
+	Prerelease bool      `json:"prerelease"`
+	CreatedAt  time.Time `json:"created_at"`
+	Assets     []struct {
+		Name               string `json:"name"`
+		BrowserDownloadURL string `json:"browser_download_url"`
+	} `json:"assets"`
+}
+
+func doUpdate(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+
+	// Create platform checker
+	platform, err := NewPlatformChecker()
+	if err != nil {
+		return err
+	}
+
+	// Get executable path
+	execPath, err := GetExecutablePath()
+	if err != nil {
+		return err
+	}
+
+	// Check if this is a managed installation
+	if platform.IsManagedInstallation(execPath) {
+		cmd.Println("This is a managed piri installation.")
+		cmd.Println("Manual updates are not supported for managed installations.")
+		cmd.Println("")
+		cmd.Println("Options:")
+		cmd.Println("  1. Enable auto-updates: sudo systemctl enable --now piri-updater.timer")
+		cmd.Println("  2. Reinstall with new version: Download new version and run 'sudo piri install --config <config>'")
+		return fmt.Errorf("cannot manually update managed installation")
+	}
+
+	// Check for updates
+	updateInfo, err := checkForUpdate(ctx, cmd)
+	if err != nil {
+		return err
+	}
+
+	if !updateInfo.NeedsUpdate {
+		cmd.Println("Already running the latest version")
+		return nil
+	}
+
+	if checkOnly {
+		cmd.Printf("Update available: %s -> %s\n", updateInfo.CurrentVersion, updateInfo.LatestVersion)
+		return nil
+	}
+
+	// Check if we need elevated privileges and handle sudo if necessary
+	if NeedsElevatedPrivileges(execPath) {
+		if !platform.IsRoot {
+			cmd.Println("Update requires administrator privileges...")
+			// re-run this exact command with sudo
```

Comment:
yea, I don't love running sudo on the users behalf. On the other hand, needs clear docs

Reply:
but my first instinct when a program I run w/o sudo asks me for my password all of a sudden is too freak out.